libs/common/src/types/mongo-db-models/tfm-deal-cancellation.ts

```diff
@@ -0,0 +1,7 @@
+import { TFM_DEAL_CANCELLATION_STATUS } from '../../constants';
+import { TfmDealCancellationWithoutStatus } from '../deal-cancellation';
+import { ValuesOf } from '../types-helper';
+
+export type TfmDealCancellationWithStatus = TfmDealCancellationWithoutStatus & {
```

Comment:
Could `TfmDealCancellation` just have an optional `status` instead of `TfmDealCancellationWithoutStatus` and `TfmDealCancellationWithStatus`?

Or perhaps it should initiate with a default `DRAFT` status and not optional?

Reply:
The `TfmDealCancellationWithoutStatus` is what the user submits and is the payload on the PUT and POST request.
The `TfmDealCancellationWithStatus` is what's stored in the database & returned on GET requests.
These are two distinct types so I dont want them to the same. I'll edit the names though

Reply:
Thanks @MarRobSoftwire , makes sense - thanks for the rename, much better than having `TfmDealCancellationWithoutStatus` and `TfmDealCancellationWithStatus` 😄